packages/ui/src/util/index.ts

```diff
@@ -48,3 +47,7 @@ export function nextIncrementalCounter(): number {
 export function uniqueElementId(): string {
   return `soraui-uid-${nextIncrementalCounter()}`
 }
+
+export function escapeStringRegexp(string: string) {
+  return string.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
+}
```

Comment:
I think it's better to use external, tested library for it, suuch as [`escape-string-regexp`](https://www.npmjs.com/package/escape-string-regexp).

Reply:
fixed